src/core/text/TextMetrics.js

```diff
@@ -0,0 +1,191 @@
+import Text from './Text';
+
+/**
+ * The TextMetrics object represents the measurement of a block of text with a specified style.
+ *
+ * @class
+ * @memberOf PIXI
+ */
+export default class TextMetrics
+{
+    /**
+     * @param {string} text - the text that was measured
+     * @param {PIXI.TextStyle} style - the style that was measured
+     * @param {number} width - the measured width of the text
+     * @param {number} height - the measured height of the text
+     * @param {array} lines - an array of the lines of text broken by new lines and wrapping if specified in style
+     * @param {array} lineWidths - an array of the line widths for each line matched to `lines`
+     * @param {number} lineHeight - the measured line height for this style
+     * @param {number} maxLineWidth - the maximum line width for all measured lines
+     * @param {Object} fontProperties - the font properties object from Text.calculateFontProperties
+     */
+    constructor(text, style, width, height, lines, lineWidths, lineHeight, maxLineWidth, fontProperties)
+    {
+        this.text = text;
+        this.style = style;
+        this.width = width;
+        this.height = height;
+        this.lines = lines;
+        this.lineWidths = lineWidths;
+        this.lineHeight = lineHeight;
+        this.maxLineWidth = maxLineWidth;
+        this.fontProperties = fontProperties;
+    }
+
+    /**
+     * Measures the supplied string of text and returns a Rectangle.
+     *
+     * @param {string} text - the text to measure.
+     * @param {PIXI.TextStyle} style - the text style to use for measuring
+     * @param {boolean} [wordWrap] - optional override for if word-wrap should be applied to the text.
+     * @param {HTMLCanvasElement} [canvas] - optional specification of the canvas to use for measuring.
+     * @return {PIXI.TextMetrics} measured width and height of the text.
+     */
+    static measure(text, style, wordWrap, canvas)
+    {
+        if (!canvas)
+        {
+            TextMetrics._canvas = TextMetrics._canvas || document.createElement('canvas');
+            canvas = TextMetrics._canvas;
+        }
+
+        wordWrap = wordWrap || style.wordWrap;
+        const font = Text.getFontStyle(style);
+        const fontProperties = Text.calculateFontProperties(font);
+        const context = canvas.getContext('2d');
```

Comment:
Could we move this to straight after getting the canvas please, as it makes sense to get the context straight after knowing what canvas you have. 

In fact, there is almost a case for passing a context for a canvas through, rather than a canvas itself, so this class isn't doing the same getting of the context over and over (when this function was in Text, it had a cached context) but I don't know how others feel about that. @bigtimebuddy?